### ISSUE TITLE
Error when deleting a layer

### ISSUE THREAD

**Issue by reckart** [inception-project/inception#2945]
**Describe the bug**
Sometimes, a layer cannot be deleted because it is still used somewhere. Normally, the UI should prevent deletion in such a case.

**To Reproduce**
Not sure, found in logs.

**Expected behavior**
If a layer cannot be deleted, the UI should not offer to delete it. Optimally, it should say why it cannot be deleted. Even if there is a bug causing the DB constraints to kick in, there should be a proper error message in the logs which says which layer the user attempted to delete so we can better figure out what's going on.

**Screenshots**
```
2022-04-04 11:42:52 ERROR ProjectSettingsPage - Error: could not execute statement; SQL [n/a]; constraint [null]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; constraint [null]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
        at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:276) ~[spring-orm-5.3.18.jar!/:5.3.18]
        at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:233) ~[spring-orm-5.3.18.jar!/:5.3.18]
        at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:566) ~[spring-orm-5.3.18.jar!/:5.3.18]
        at org.springframework.transaction.support.AbstractPlatformTransactionManager.processCommit(AbstractPlatformTransactionManager.java:743) ~[spring-tx-5.3.18.jar!/:5.3.18]
        at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:711) ~[spring-tx-5.3.18.jar!/:5.3.18]
        at org.springframework.transaction.interceptor.TransactionAspectSupport.commitTransactionAfterReturning(TransactionAspectSupport.java:654) ~[spring-tx-5.3.18.jar!/:5.3.18]
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:407) ~[spring-tx-5.3.18.jar!/:5.3.18]
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119) ~[spring-tx-5.3.18.jar!/:5.3.18]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.18.jar!/:5.3.18]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753) ~[spring-aop-5.3.18.jar!/:5.3.18]
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:698) ~[spring-aop-5.3.18.jar!/:5.3.18]
        at de.tudarmstadt.ukp.clarin.webanno.api.dao.AnnotationSchemaServiceImpl$$EnhancerBySpringCGLIB$$26024283.removeLayer(<generated>) ~[inception-api-dao-23.0.1.jar!/:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.apache.wicket.proxy.jdk.JdkProxyFactory$JdkHandler.invoke(JdkProxyFactory.java:172) ~[wicket-ioc-9.7.0.jar!/:9.7.0]
        at com.sun.proxy.$Proxy218.removeLayer(Unknown Source) ~[?:?]
        at de.tudarmstadt.ukp.clarin.webanno.ui.project.layers.LayerDetailForm.lambda$actionDelete$59c46454$1(LayerDetailForm.java:331) ~[inception-ui-project-23.0.1.jar!/:?]
```

**Please complete the following information:**
 - Version and build ID: [see bottom of the browser screen]
 - OS: [e.g. Windows, Linux, OS X]
 - Browser: [e.g. chrome, safari]

**Additional context**

**Comment by reckart** [comment]
Could be a duplicate of https://github.com/inception-project/inception/pull/2951 or at least triggered by it.

**Comment by reckart** [comment]
Fixed with #3081 - the problematic layer was `Lemma` and the problem was caused by it having been wrongly hooked up to the `Token.morph` feature.
